src/App.js

```diff
@@ -1,16 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatMessagesData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedMessage) => {
+    const messages = chatMessagesData.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+
+    });
+    setChatData(messages);
+  };
+
+
+  const countLikes = (messages) => {
+    let likes = 0;
+    messages.forEach((message) => {
+      if (message.liked) {
+        likes++;
+      }
+    });
+    return likes;
+  };
+
+  const likes = countLikes(chatMessagesData);
+
   return (
     <div id="App">
+
       <header>
-        <h1>Application title</h1>
+        <h1>A tale of two cities</h1>
+        <section><span id="heartWidget" className="widget"> {likes} 💚s</span></section>
```

Comment:
You could just call your function in jsx like `{countLikes(chatMessageData}` and in Learn you were getting an error message because of the style of heart you are using. You can alter the tests to accept the heart you want to use.